### PR TITLE
[MERGE] (event)(_sale)(_product_configurator): clean up

### PR DESCRIPTION
**Pull request by Feyensv**
1* Allow products access from product_id / product_template_id of sale.order.line
2* Allow use of optional products along rentals and events.

### CODE REVIEW

File: addons/sale_product_configurator/static/src/js/product_configurator_widget.js [hunk 1]
```diff
@@ -23,12 +23,16 @@ ProductConfiguratorWidget.include({
 
     /**
      * Set restoreProductTemplateId for further backtrack.
```
Comment (dbo-odoo) [inline addons/sale_product_configurator/static/src/js/product_configurator_widget.js]:
```javascript
/**
* Set restoreProductTemplateId for further backtrack. Sets the options on the line for future steps.
*/
```

File: addons/sale_product_configurator/static/src/js/product_configurator_widget.js [hunk 2]
```diff
@@ -322,28 +316,28 @@ ProductConfiguratorWidget.include({
             };
 
             if (product.no_variant_attribute_values) {
-                var default_product_no_variant_attribute_values = [];
-                _.each(product.no_variant_attribute_values, function (attribute_value) {
-                        default_product_no_variant_attribute_values.push(
-                            [4, parseInt(attribute_value.value)]
+                var defaultProductNoVariantAttributeValues = [];
```
Comment (dbo-odoo) [inline addons/sale_product_configurator/static/src/js/product_configurator_widget.js]: `VeryInterestingDiff`!
`or_is_it`?

File: addons/sale/static/src/js/product_configurator_widget.js [hunk 3]
```diff
@@ -134,12 +160,56 @@ var ProductConfiguratorWidget = relationalFields.FieldMany2One.extend({
...
+     *        the result of the super call for other wizard extensions
+     *    false if the product wasn't configurable through the wizard
+     *
+     * @param {String} dataPointID
+     * @returns {Promise<Boolean>} stopPropagation true if a suitable configurator has been found.
+     *
+     * @private
+     */
+    _onLineConfigured: function () {
+
```
Comment (dbo-odoo) [inline addons/sale/static/src/js/product_configurator_widget.js]: the base implementation does not respect your spec in the docstring

File: addons/sale_product_configurator/static/src/js/product_configurator_widget.js (hunk 2 above)
Comment (Feyensv) [reply addons/sale_product_configurator/static/src/js/product_configurator_widget.js]: I took the opportunity to pass the js through a more thorough linter check, because in any case this code is mine, but i can drop it if you really don't want it :D.

**Comment by dbo-odoo** [comment]
It can stay, I just couldn't help but notice is allOn Jul 8, 2019 8:16 PM, vfe <notifications@github.com> wrote:@Feyensv commented on this pull request.



In addons/sale_product_configurator/static/src/js/product_configurator_widget.js:
> @@ -322,28 +316,28 @@ ProductConfiguratorWidget.include({
             };
 
             if (product.no_variant_attribute_values) {
-                var default_product_no_variant_attribute_values = [];
-                _.each(product.no_variant_attribute_values, function (attribute_value) {
-                        default_product_no_variant_attribute_values.push(
-                            [4, parseInt(attribute_value.value)]
+                var defaultProductNoVariantAttributeValues = [];

I took the opportunity to pass the js through a more thorough linter check, because in any case this code is mine, but i can drop it if you really don't want it :D.

File: addons/sale_product_configurator/static/src/js/product_configurator_widget.js [hunk 4]
```diff
@@ -121,15 +137,15 @@ ProductConfiguratorWidget.include({
      * @private
      */
     _openProductConfigurator: function (data, dataPointId) {
+        this.options = undefined;
```
Comment (dbo-odoo) [inline addons/sale_product_configurator/static/src/js/product_configurator_widget.js]: why is this needed?

File: addons/sale_product_configurator/static/src/js/tours/sale_product_configurator.js [hunk 5]
```diff
@@ -1,18 +0,0 @@
-odoo.define("sale_product_configurator.tour", function (require) {
-"use strict";
-/**
```
Comment (dbo-odoo) [inline addons/sale_product_configurator/static/src/js/tours/sale_product_configurator.js]: why is this no longer needed?

Comment (Feyensv) [reply addons/sale_product_configurator/static/src/js/tours/sale_product_configurator.js]: The sale tour was fixed to take either the product_id or product_template_id for its tour.
So this "hack" isn't needed anymore

File: addons/sale_product_configurator/static/src/js/product_configurator_widget.js (hunk 4 above)
Comment (Feyensv) [reply addons/sale_product_configurator/static/src/js/product_configurator_widget.js]: It ensures the options are reset when we reOpen the Configurator and that they cannot be applied twice.  I don't think it can happen currently but it is a layer of security :3.

**Comment by dbo-odoo** [comment]
@robodoo r+ rebase-merge
finger crossed
